relay-server/src/utils/dynamic_sampling.rs

```diff
@@ -0,0 +1,441 @@
+//! Functionality for calculating if a trace should be processed or dropped.
+//!
+use std::convert::TryInto;
+
+use rand::{distributions::Uniform, Rng};
+use rand_pcg::Pcg32;
+use serde::{Deserialize, Serialize};
+
+use relay_common::{ProjectId, ProjectKey, Uuid};
+use relay_filter::GlobPatterns;
+
+/// A sampling rule defined by user in Organization options
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct SamplingRule {
+    /// The project ids for which the sample rule applies, empty applies to all
+    #[serde(default)]
+    pub project_ids: Vec<ProjectId>,
+    /// A set of Glob patterns for which the rule applies, empty applies to all
+    #[serde(default)]
+    pub releases: GlobPatterns,
```

Comment:
I'm realizing that our proposal included releases but not environments. Please also add environments here.

Reply:
Ok, I'll get back to you on this.